code/game/machinery/tcomms/core.dm

```diff
@@ -257,9 +266,35 @@
 			to_chat(usr, "<span class='notice'>Successfully changed password from <b>[link_password]</b> to <b>[new_password]</b>.</span>")
 			link_password = new_password
 
+	if(ui_tab == UI_TAB_FILTER)
+		if(href_list["add_filter"])
+			// This is a stripped input because I did NOT come this far for this system to be abused by HTML injection
+			var/name_to_add = stripped_input(usr, "Enter a name to add to the filtering list", "Name Entry")
+			if(name_to_add == "")
+				return
+			if(name_to_add in nttc.filtering)
+				to_chat(usr, "<span class='alert'><b>ERROR:</b> User already in filtering list.</span>")
+			else
+				nttc.filtering |= name_to_add
+				log_action(usr, "has added [name_to_add] to the NTTC filter list on core with ID [network_id]", TRUE)
```

Comment:
Side thing. But can you add logging using the new logging system to the log_action proc?
So just `user.create_log(...)`

Reply:
This is a specific proc to NTTC with flags of whether it should admin log. If you wish, you can modify `log_action` directly to add that in.